adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/CoverageRDD.scala

```diff
@@ -59,17 +62,52 @@ private[adam] class CoverageArraySerializer(kryo: Kryo) extends IntervalArraySer
   }
 }
 
-object CoverageRDD {
+case class ParquetUnboundCoverageRDD private[rdd] (
+    private val sc: SparkContext,
+    private val parquetFilename: String,
+    sequences: SequenceDictionary) extends CoverageRDD {
 
-  /**
-   * Builds a CoverageRDD that does not have a partition map.
-   *
-   * @param rdd The underlying Coverage RDD.
-   * @param sd The SequenceDictionary for the RDD.
-   * @return A new Coverage RDD.
-   */
-  def apply(rdd: RDD[Coverage], sd: SequenceDictionary): CoverageRDD = {
-    CoverageRDD(rdd, sd, None)
+  lazy val rdd: RDD[Coverage] = {
+    sc.loadParquetCoverage(parquetFilename,
+      forceRdd = true).rdd
+  }
+
+  protected lazy val optPartitionMap = sc.extractPartitionMap(parquetFilename)
+
+  lazy val dataset = {
+    val sqlContext = SQLContext.getOrCreate(sc)
+    import sqlContext.implicits._
+    sqlContext.read.parquet(parquetFilename)
+      .select("contigName", "start", "end", "score")
+      .withColumnRenamed("score", "count")
+      .as[Coverage]
+  }
+
+  def toFeatureRDD(): FeatureRDD = {
+    ParquetUnboundFeatureRDD(sc, parquetFilename, sequences)
```

Comment:
I'm a pretty strong -1 on that. It's hard to implement a performant caching strategy without a good model for the size of the unserialized data in memory, the total resource constraints of the cluster, or knowledge of the full compute DAG. Even for two pass algorithms like BQSR and INDEL realignment, caching isn't always a win. Yes, you don't re-read from disk, but if you've got a fast file system, the memory backpressure may not outperform the I/O savings, you may not have sufficient space to cache the whole deserialized RDD, etc.